### PR TITLE
Implementation of the HBaseDDLExecutorContext.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-7685

### CODE REVIEW

**Comment by sagarkapare** [comment]
Build is running here - https://builds.cask.co/browse/CDAP-DUT5353-1

File: cdap-hbase-spi/src/main/java/co/cask/cdap/spi/hbase/HBaseDDLExecutorContext.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * HBase DDL executor context.
+ */
+public interface HBaseDDLExecutorContext {
+  /**
+   * Returns the hadoop conf with which the {@link HBaseDDLExecutor} was initialized.
+   * @param <T>
+   * @return the conf
+   */
+  <T> T getConf();
```
Comment (albertshau) [inline cdap-hbase-spi/src/main/java/co/cask/cdap/spi/hbase/HBaseDDLExecutorContext.java]: let's call this getConfiguration to make it even more clear that it's the Hadoop Configuration class

File: cdap-hbase-spi/src/main/java/co/cask/cdap/spi/hbase/HBaseDDLExecutorContext.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  /**
+   * Returns the {@link Map} of properties associated with the HBase DDL executor extension.
```
Comment (albertshau) [inline cdap-hbase-spi/src/main/java/co/cask/cdap/spi/hbase/HBaseDDLExecutorContext.java:33]: should mention that this contains anything in the cdap config that is prefixed with 'cdap.hbase.spi.', but with 'cdap.hbase.spi.' stripped.

File: cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java [hunk 3]
```diff
@@ -434,12 +433,13 @@ public static void setVersion(HTableDescriptorBuilder tableDescriptorBuilder) {
...
    * @param namespaceId namespace for which the tables are being deleted
+   * @param hConf The {@link Configuration} instance
    * @param predicate The {@link Predicate} to decide whether to drop a table or not
    * @throws IOException
    */
   public void deleteAllInNamespace(HBaseDDLExecutor ddlExecutor, String namespaceId,
-                                   Predicate<TableId> predicate) throws IOException {
-    HBaseAdmin admin = new HBaseAdmin((Configuration) ddlExecutor.getConf());
+                                   Configuration hConf, Predicate<TableId> predicate) throws IOException {
+    HBaseAdmin admin = new HBaseAdmin(hConf);
```
Comment (albertshau) [inline cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java]: this is already a problem, but this is never closed

**Review by albertshau — COMMENTED** [review]
couple minor comments

File: cdap-hbase-spi/src/main/java/co/cask/cdap/spi/hbase/HBaseDDLExecutorContext.java (hunk 1 above)
Comment (sagarkapare) [reply cdap-hbase-spi/src/main/java/co/cask/cdap/spi/hbase/HBaseDDLExecutorContext.java]: sounds good. fixed.

File: cdap-hbase-spi/src/main/java/co/cask/cdap/spi/hbase/HBaseDDLExecutorContext.java (hunk 2 above)
Comment (sagarkapare) [reply cdap-hbase-spi/src/main/java/co/cask/cdap/spi/hbase/HBaseDDLExecutorContext.java:33]: updated the documentation.

File: cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java (hunk 3 above)
Comment (sagarkapare) [reply cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java]: oh yes. fixed.

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by sagarkapare** [comment]
Build is green. Squashing and merging.

**Pull request merged**